### ISSUE TITLE
FR: Easier to navigate (and complete) API reference

### ISSUE THREAD

**Issue by rhofour** [joanllenas/ts.data.json#20]
This library would be easier to use if you had an autogenerated API reference that makes it easy to see all the functions available.

For example, I just discovered that both Err<A> and Ok<A> have an isOk method by reading the code. These methods are useful and it'd be nice if they were easier to find. That said, the code is very readable from what little I've seen.

**Comment by joanllenas** [comment]
You are right, something that generates docs automatically would be much better. Any recommendations? ideally something compatible with Github pages, wikis or the like. If I ever do this I'd like to automate it fully.

On the other hand, that particular method `isOk` is an implementation detail. I wouldn't recommend relying on those. I would rely on the _unwraping_ functions, which are not documented 😅  There's another issue related to improving those: https://github.com/joanllenas/ts.data.json/issues/19
I plan on doing that one soon, maybe I'll tackle both at the same time.

**Comment by rhofour** [comment]
Unfortunately I haven't really touched any web stuff in many years so I have no idea what existing solutions are out there, but I imagine there are a few options.

**Comment by joanllenas** [comment]
I found something interesting: https://typedoc.org/
I'll give it a try when I have some time.
